src/lib/components/themetoggle/main.svelte

```diff
@@ -44,18 +63,14 @@
 	<meta name="darkreader-lock" />
 	<!-- Load theme before body starts rendering. -->
 	<script>
-		const selectedTheme = localStorage.getItem('theme') ?? 'system';
-		const systemTheme = window.matchMedia('(prefers-color-scheme: dark)').matches
-			? 'dark'
-			: 'light';
-		const theme = selectedTheme !== 'system' ? selectedTheme : systemTheme;
-		if (theme === 'dark') {
-			document.documentElement.classList.add('dark');
-		}
+		const selected = localStorage.getItem('theme') ?? 'system';
+		const system = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+		const theme = selected === 'system' ? system : selected;
+		if (theme === 'dark') document.documentElement.classList.add('dark');
 	</script>
 </svelte:head>
 
-<button class="absolute top-0 right-0 md:p-4" onclick={toggleTheme}>
+<button class="absolute top-0 right-0 md:p-4" onclick={toggleTheme} ondblclick={clearTheme}>
```

Comment:
This could probably be better with "on hold" but that needs to be made manually with interval, since we are planning to move to a settings modal I took the liberty of being lazy here :)